flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/Lock.java

```diff
@@ -31,13 +32,57 @@ public interface Lock extends AutoCloseable {
     /** Run with lock. */
     <T> T runWithLock(Callable<T> callable) throws Exception;
 
-    @Nullable
-    static Lock fromCatalog(CatalogLock.Factory lockFactory, ObjectPath tablePath) {
-        if (lockFactory == null) {
-            return null;
+    /** A factory to create {@link Lock}. */
+    interface Factory extends Serializable {
+        Lock create();
+    }
+
+    static Factory factory(@Nullable CatalogLock.Factory lockFactory, ObjectPath tablePath) {
+        return lockFactory == null
+                ? new EmptyFactory()
+                : new CatalogLockFactory(lockFactory, tablePath);
+    }
+
+    static Factory emptyFactory() {
+        return new EmptyFactory();
+    }
+
+    /** A {@link Factory} creating lock from catalog. */
+    class CatalogLockFactory implements Factory {
+
+        private static final long serialVersionUID = 1L;
+
+        private final CatalogLock.Factory lockFactory;
+        private final ObjectPath tablePath;
+
+        public CatalogLockFactory(CatalogLock.Factory lockFactory, ObjectPath tablePath) {
+            this.lockFactory = lockFactory;
+            this.tablePath = tablePath;
+        }
+
+        @Override
+        public Lock create() {
+            return fromCatalog(lockFactory.create(), tablePath);
         }
+    }
 
-        return fromCatalog(lockFactory.create(), tablePath);
+    /** A {@link Factory} creating empty lock. */
+    class EmptyFactory implements Factory {
+
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public Lock create() {
+            return new Lock() {
+                @Override
+                public <T> T runWithLock(Callable<T> callable) throws Exception {
+                    return callable.call();
+                }
+
+                @Override
+                public void close() {}
+            };
```

Comment:
Just return `null`?

Reply:
I think we can return a object to avoid null checking.
I will modify others to remove null checking.